textile/chat-features.textile

```diff
@@ -413,12 +416,16 @@ Typing Indicators allow chat room users to indicate to others that they are typi
 * @(CHA-T3)@ @[Testable]@ Users may configure a timeout interval for when they are typing. This configuration is provided as part of the @RoomOptions@ @typing.timeoutMs@ property, or idiomatic equivalent. The default is 5000ms.
 * @(CHA-T4)@ Users may indicate that they have started typing.
 ** @(CHA-T4a)@ @[Testable]@ If typing is not already in progress, per explicit cancellation or the timeout interval in (@CHA-T3@), then a new typing session is started.
-*** @(CHA-T4a1)@ @[Testable]@ When a typing session is started, the client is entered into presence on the typing channel.
-*** @(CHA-T4a2)@ @[Testable]@ When a typing session is started, a timeout is set according to the @CHA-T3@ timeout interval. When this timeout expires, the typing session is automatically ended by leaving presence.
+*** @(CHA-T4a3)@ @[Testable]@ If the room status is @Attaching@, then the operation shall invoke a @CHA-RL9@ pending room status operation and wait for it to complete. If this operation fails, then the error from @CHA-RL9@ must be raised. If this succeeds (i.e the room becomes @ATTACHED@), the library will proceed per @CHA-T4a1@.
+*** @(CHA-T4a4)@ @[Testable]@ If the room status is anything other than @Attached@ or @Attaching@, an @ErrorInfo@ using the @RoomInInvalidState@ error code from the "chat-specific error codes":#error-codes with a status code of @400@ must be thrown. The message shall explain that attach must be called first.
+*** @(CHA-T4a1)@ @[Testable]@ If the room status is @Attached@, then the @start@ call will invoke the underlying @presence.enter()@ call.
+*** @(CHA-T4a2)@ @[Testable]@ When a typing session is started per @CHA-T4a1@, a timeout is set according to the @CHA-T3@ timeout interval. When this timeout expires, the typing session is automatically ended by leaving presence.
```

Comment:
```suggestion
*** @(CHA-T4a2)@ @[Testable]@ When a typing session is started per @CHA-T4a1@, a timeout is set according to the @CHA-T3@ timeout interval. When this timeout expires, the typing session automatically ends by leaving presence as per @CHA-T5c@.
```

Reply:
Not sure, if we should have `internal retry mechanism` to stop sending typing event if `presence.leave` fails.
Since, we will not be delegating this error to the user : (

Reply:
Yes - this is a known bug that we intend to fix at a later date, I would prefer not to spec for it until we know what that is

Reply:
if implicit `leave` event fails due to `typing timeout`, receivers will see eternal `typing` in progress : (

Reply:
Yes - it's a known bug. Once someone fixes it in a satisfactory way (in any of the SDKs), we'll update the spec - I just don't want to write a spec point for it until we have a tried and tested fix :)

Reply:
If we are tracking it somewhere then good, otherwise you might like to create separate github issue for this WDYT

Reply:
Just an idea,
- All implicit ops that don't notify user about the error are retried for max_retries = 3 and then added to the internal queue of implicit failed operations.
- All those operations will be performed again once the connection goes `connected`.

Reply:
Seems as per [RTP16b](https://sdk.ably.com/builds/ably/specification/main/features/#RTP16b), presence events will be queued at channel level when channel in `Initialized` or `Attaching` state.

Reply:
For other states, operation will result in error as per [RTP16c](https://sdk.ably.com/builds/ably/specification/main/features/#RTP16c)

Reply:
You might like to add those points to JIRA ticket, I will approve the PR for now, to unblock the spec